odoo/fields.py

```diff
@@ -2583,6 +2583,44 @@ def convert_to_display_name(self, value, record):
         return ustr(value and value.display_name)
 
 
+class Domain(Field):
+    """
+    Field that contain a domain on a given model
+
+    :param str domain_model: name of the model on which the domain apply.
+    :param str domain_model_field: name of the :Class:`Char`, :class:Selection, or :class:`Many2one` where the model name is stored.
```

Comment:
I haven't seen any use case for `domain_model`.
And `fields.Domain(domain_model_field="...")` looks clunky to me.
What about `fields.Domain(model_field="...")`, like the `Many2oneReference` field?

Reply:
There is some cases where the model is static: 
  - [gamification.challenge/user_domain](https://github.com/odoo/odoo/blob/e7644618522d34872b0b8ba81cff1e84217f88d1/addons/gamification/models/challenge.py#L76): always apply on res.users
  - [pos.cache/product_domain](https://github.com/odoo/odoo/blob/e7644618522d34872b0b8ba81cff1e84217f88d1/addons/pos_cache/models/pos_cache.py#L16): always product.product
  - [coupon.rule/rule_partners_domain and coupon.rule/rule_product_domain](https://github.com/odoo/odoo/blob/e7644618522d34872b0b8ba81cff1e84217f88d1/addons/coupon/models/coupon_rules.py#L14-L15): always on res.partner and product.product.
  - [documents.workflow.rule/domain](https://github.com/odoo/enterprise/blob/6c71a54cc0b6fca883dc281ec24fc09421b3753c/documents/models/workflow.py#L24): [always on documents](https://github.com/odoo/enterprise/blob/6c71a54cc0b6fca883dc281ec24fc09421b3753c/documents/models/document.py#L194-L211)
  - ...

And yes, the attribute can be renamed. The "field" version can also be the default one (without kwargs)